integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonAddLoadCommand.scala

```diff
@@ -93,7 +93,14 @@ case class CarbonAddLoadCommand(
 
     // If a path is already added then we should block the adding of the same path again.
     val allSegments = SegmentStatusManager.readLoadMetadata(carbonTable.getMetadataPath)
-    if (allSegments.exists(a => a.getPath != null && a.getPath.equalsIgnoreCase(inputPath))) {
+    // If the segment has been already loaded from the same path and its status is SUCCESS or
+    // PARTIALLY_SUCCESS, throw an exception as we should block the adding of the same path again.
+    if (allSegments.exists(a => a.getPath != null && a.getPath.equalsIgnoreCase(inputPath) &&
+                                (a.getSegmentStatus.getMessage
+                                   .equalsIgnoreCase(SegmentStatus.SUCCESS.toString) ||
```

Comment:
Can directly check for `a.getSegmentStatus == SegmentStatus.SUCCESS || a.getSegmentStatus == 
SegmentStatus .LOAD_PARTIAL_SUCCESS`

Reply:
format the code properly

Reply:
done